pkg/util/test/util.go

```diff
@@ -2578,6 +2610,255 @@ func ValidateAlertManagerDisabled(pxImageList map[string]string, cluster *corev1
+	if err != nil {
+		return err
+	}
+
+	// Verify registration container image inside px-telemetry-registration
+	telemetryImageName, ok := pxImageList["telemetry"]
+	if !ok {
+		return fmt.Errorf("failed to find image for telemetry")
+	}
+	telemetryImageName = util.GetImageURN(cluster, telemetryImageName)
+
+	if registrationServiceDep.Spec.Template.Spec.Containers[0].Image != telemetryImageName {
+		return fmt.Errorf("registration image mismatch, image: %s, expected: %s",
+			registrationServiceDep.Spec.Template.Spec.Containers[0].Image,
+			telemetryImageName)
+	}
+	// Verify envoy container image inside registration-service
+	envoyImageName, ok := pxImageList["telemetryProxy"]
+	if !ok {
+		return fmt.Errorf("failed to find image for envoy")
+	}
+	envoyImageName = util.GetImageURN(cluster, envoyImageName)
+	if registrationServiceDep.Spec.Template.Spec.Containers[1].Image != envoyImageName {
+		return fmt.Errorf("envoy image mismatch, image: %s, expected: %s",
+			registrationServiceDep.Spec.Template.Spec.Containers[1].Image,
+			envoyImageName)
+	}
+
+	// Verify px-telemetry-metrics-collector deployment and images
+	logrus.Info("Validate px-telemetry-metrics-collector deployment and images")
+	metricsCollectorDep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-telemetry-metrics-collector",
+			Namespace: cluster.Namespace,
+		},
+	}
+	if err := appops.Instance().ValidateDeployment(metricsCollectorDep, timeout, interval); err != nil {
+		return err
+	}
+	metricsCollectorDep, err = appops.Instance().GetDeployment(metricsCollectorDep.Name, metricsCollectorDep.Namespace)
+	if err != nil {
+		return err
+	}
+	// Verify collector container image inside px-metrics-collector
+	collectorImageName, ok := pxImageList["metricsCollector"]
+	if !ok {
+		return fmt.Errorf("failed to find image for metrics collector")
+	}
+	collectorImageName = util.GetImageURN(cluster, collectorImageName)
+
+	if metricsCollectorDep.Spec.Template.Spec.Containers[0].Image != collectorImageName {
+		return fmt.Errorf("collector image mismatch, image: %s, expected: %s",
+			metricsCollectorDep.Spec.Template.Spec.Containers[0].Image,
+			collectorImageName)
+	}
+	// Verify envoy container image inside px-metrics-collector
+	if metricsCollectorDep.Spec.Template.Spec.Containers[1].Image != envoyImageName {
+		return fmt.Errorf("envoy image mismatch, image: %s, expected: %s",
+			metricsCollectorDep.Spec.Template.Spec.Containers[1].Image,
+			envoyImageName)
+	}
+
+	// Wait for the daemonset to become online
+	logrus.Info("Validate px-telemetry-phonehome daemonset and images")
+	if err := appops.Instance().ValidateDaemonSet("px-telemetry-phonehome", cluster.Namespace, timeout); err != nil {
+		return err
+	}
+	telemetryPhonehomeDs, err := appops.Instance().GetDaemonSet("px-telemetry-phonehome", cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Verify collector container image inside px-telemetry-phonehome
```

Comment:
same here, the template should have an init container, we can just check the image.

Reply:
Done